derive/src/lib.rs

```diff
@@ -45,15 +45,14 @@ fn expand(input: DeriveInput) -> Result<TokenStream2> {
     };
 
     Ok(quote! {
-        impl #impl_generics ::ref_cast::RefCast for #name #ty_generics #where_clause {
-            type From = #from;
+        impl #impl_generics ::ref_cast::RefCast<#from> for #name #ty_generics #where_clause {
 
             #[inline]
-            fn ref_cast(_from: &Self::From) -> &Self {
-                // TODO: assert that `Self::From` and `Self` have the same size
+            fn ref_cast(_from: &#from) -> &Self {
+                // TODO: assert that `#from` and `Self` have the same size
```

Comment:
I don't think this changes things with regards to this comment, but I'm not sure of all the concrete cases where it causes problems...